pkg/kclient/jobs.go

```diff
@@ -0,0 +1,87 @@
+package kclient
+
+import (
+	"context"
+	"fmt"
+	"io"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog"
+)
+
+// constants for volumes
+const (
+	JobsKind       = "Job"
+	JobsAPIVersion = "batch/v1"
+)
+
+func (c *Client) ListJobs(selector string) (*batchv1.JobList, error) {
+	return c.KubeClient.BatchV1().Jobs(c.Namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: selector})
+}
+
+// CreateJobs creates a K8s job to execute task
+func (c *Client) CreateJob(job batchv1.Job, namespace string) (*batchv1.Job, error) {
+	if namespace == "" {
+		namespace = c.Namespace
+	}
+	createdJob, err := c.KubeClient.BatchV1().Jobs(namespace).Create(context.TODO(), &job, metav1.CreateOptions{FieldManager: FieldManager})
+	if err != nil {
+		return nil, fmt.Errorf("unable to create Jobs: %w", err)
+	}
+	return createdJob, nil
+}
+
+// WaitForJobToComplete to wait until a job completes or fails; it starts printing log or error if the job does not complete execution after 2 minutes
+func (c *Client) WaitForJobToComplete(job *batchv1.Job) (*batchv1.Job, error) {
+	klog.V(3).Infof("Waiting for Job %s to complete successfully", job.Name)
+
+	w, err := c.KubeClient.BatchV1().Jobs(c.Namespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector: fields.Set{"metadata.name": job.Name}.AsSelector().String(),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("unable to watch job: %w", err)
+	}
+	defer w.Stop()
+
+	for {
+		val, ok := <-w.ResultChan()
+		if !ok {
+			break
+		}
+		wJob := val.Object.(*batchv1.Job)
```

Comment:
This could panic if the result is of type metav1.Status in case of error (and, if I understand correctly, the error won't be covered by the ok checked previously)